datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeFlowEventChainFactory.java

```diff
@@ -71,4 +79,33 @@ public FlowTriggerEventQueue createFlowTriggerEventQueue(DatalakeResizeFlowChain
 
         return new FlowTriggerEventQueue(getName(), event, chain);
     }
+
+    private void addFreeIpaUpscaleToChain(Queue<Selectable> chain, DatalakeResizeFlowChainStartEvent event, String envCrn, boolean accept) {
+        if (accept) {
+            chain.add(new FreeIpaUpscaleStartEvent(null, event.getResourceId(), event.getUserId(), envCrn, event.accepted()));
+        } else {
+            chain.add(new FreeIpaUpscaleStartEvent(event.getResourceId(), event.getUserId(), envCrn));
+
+        }
```

Comment:
this is not covered with unit test

Reply:
At this point accept is aways true, I've included this case for a future where upscale is no longer the first action in the chain.